### PR TITLE
Label - Add the ability to justify text

### LINKED ISSUE #6009: [Enhancement] Ability to justify text for Label
## Summary
Word offers 4 options to align text, but Forms has 3.

Capture

> When you justify text, space is added between words so that both edges of each line are aligned with both margins. The last line in the paragraph is aligned left.

It'd be great to have the ability to justify align text.

## API Changes

Possibly adding `TextAlignment.Justify`, but this would also be present for vertical alignment. If justify doesn't make sense for vertical alignment, we can perhaps introduce a platform specific or throw an exception when `Justify` is selected. I'm leaning towards the latter to not complicate how PS and existing alignment options work, but not sure what the team thinks.

### PR DESCRIPTION
**Pull request by Seuleuzeuh**
### Description of Change ###

Add Ability to justify text for label.
Adding TextAlignment.Justify

### Issues Resolved ### 

- fixes #6009

### API Changes ###

Added :
- TextAlignment.Justify

When TextAlignement.Justify is used in VerticalAlignment property, that throw an NotSupportedException like @adrianknight89 suggested. Feel free to ask me to change this behavior.

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS
- Android ( > API 25, TextAlignment.Start either)
- UWP
- MacOS
- WPF

### Behavioral/Visual Changes ###
Ability to justify text in label

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###
I've added a test control in ControlGallery (Label & LabelLegacy)

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Controls/CoreGalleryPages/LabelCoreGalleryPage.cs [hunk 1]
```diff
@@ -9,21 +9,21 @@ protected override bool SupportsFocus
 			get { return false; }
 		}
 
-		protected override void InitializeElement (Label element)
```
Comment (MathewLC) [inline Xamarin.Forms.Controls/CoreGalleryPages/LabelCoreGalleryPage.cs]: I think that is too much unrelated change. What about you remove this changes and make it cleaner to the team review it?

**Comment by MathewLC** [comment]
@Seuleuzeuh any update on this? Let me know if I can help you someway.

**Comment by MathewLC** [comment]
@Seuleuzeuh any update on this??? I want this functionality as soon as possible because I have a lot scenarios in my app that I could use it.

**Comment by Seuleuzeuh** [comment]
> @Seuleuzeuh any update on this??? I want this functionality as soon as possible because I have a lot scenarios in my app that I could use it.

Sorry, i'm busy for the moment, i've added you to the collaborators of the repo corresponding to this PR, if you want to do the updates. 
If you can't, i'll try to do it this day.

**Comment by MathewLC** [comment]
All right, I'll try to do something about the cleaning on the weekend, but if you can do it first, I won't apply anything. Let's work together!!! :smile:

**Comment by MathewLC** [comment]
@Seuleuzeuh can you confirm that all the changes were really made? I tried to keep only the related changes. If you confirm that everything is fine, we can set up samhouts to move foward.

**Comment by MathewLC** [comment]
Is these already released on some version of nuget package? Because when it do, the 9539 can be closed too. Please, let me know. @rmarinho @samhouts

**Comment by MathewLC** [comment]
@samhouts Maybe the team can proceed  replying us if there's something we need to change to merge this as fast as possible. Best regards :blush:

**Comment by Seuleuzeuh** [comment]
> @samhouts Maybe the team can proceed replying us if there's something we need to change to merge this as fast as possible. Best regards 😊

Solution does not compile, missing coma in [file TextAlignment.cs](https://github.com/xamarin/Xamarin.Forms/commit/298054d40a78c5d7ff54595a6d38e1bb013774f1#diff-32841b2c2db3028bf17d2ab723861aff). 
My local solution is not fully operationnal, i can't test to check

**Comment by MathewLC** [comment]
I'll check as soon as possible, if you could please send me a print of the message in my personal email. Best regards. @Seuleuzeuh

**Comment by MathewLC** [comment]
I don't know what happened. When I was doing my commit github stop working, i don't if was that but the pull request got mixed with others unrelated commits. Now, although has others commits in conversation, the pull request has only related commits.  😅😅
Best regards. :satisfied:
@StephaneDelcroix  @jsuarezruiz

**Comment by MathewLC** [comment]
@jsuarezruiz just a pin. Any update on this?

File: Xamarin.Forms.Controls/GalleryPages/LabelGallery.cs [hunk 2]
```diff
@@ -13,6 +13,13 @@ public LabelGallery ()
 			var normal = new Label { Text = "Normal Label" };
 			var center = new Label { Text = "Center Label" };
 			var right = new Label { Text = "Right Label" };
+			var justify = new Label
+			{
+				HorizontalOptions = LayoutOptions.FillAndExpand,
+				Text = "Justify Label, with huge text : " +
+				"In a hole in the ground there lived a hobbit. " +
```
Comment (samhouts) [inline Xamarin.Forms.Controls/GalleryPages/LabelGallery.cs:20]:
```suggestion
				"The five boxing wizards jump quickly. " +
```

File: Xamarin.Forms.Controls/GalleryPages/LabelGallery.cs [hunk 3]
```diff
@@ -13,6 +13,13 @@ public LabelGallery ()
...
+				"Not a nasty, dirty, wet hole, filled with the ends of worms and aa oozy smell, not yet a dry, bare, sandy hole with...",
```
Comment (samhouts) [inline Xamarin.Forms.Controls/GalleryPages/LabelGallery.cs:21]:
```suggestion
				"The quick brown fox jumps over the lazy dog. How vexingly quick daft zebras jump!",
```

**Comment by MathewLC** [comment]
I'm sorry for not helping more. But now that I'm saw this, just to be sure, @PureWeen  requested changes but they were already been made by @rmarinho right? Is there anything else that needs to be done?

**Comment by MathewLC** [comment]
@samhouts can you confirm if the errors was because of something left behind by merge?
Because if it's I can do a rebase and try to remove these inconsistencies. @rmarinho @PureWeen

**Comment by MathewLC** [comment]
@rmarinho  @samhouts  @PureWeen  I tried to rebase from main as it's the actual base branch but I'm getting an error related to Ellipse reference:
``` Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0234	The type or namespace name 'Ellipse' does not exist in the namespace 'Xamarin.Forms' (are you missing an assembly reference?)	Xamarin.Forms.Controls	C:\Repos\Xamarin Fix-6009\Xamarin.Forms\Xamarin.Forms.Controls\obj\Debug\netstandard2.0\GalleryPages\ShapesGalleries\EllipseGallery.xaml.g.cs	20	Active
```

If you guys could please take a look, I'll aprecciate it. Thank in advance.

**Comment by jfversluis** [comment]
This will not happen anymore for Xamarin.Forms. If this is still important to you, make sure to check the [.NET MAUI repo](https://github.com/dotnet/maui) and see if it's already on the roadmap or open an issue with a detailed feature request. Thanks!
